### ISSUE TITLE
Many running balloon helper threads and high cpu usage

### ISSUE THREAD

**Issue by raffis** [gyselroth/balloon-client-desktop#34]
#### Description
**Is**:
~70 balloon helpers running and various errors in sync log.

sync log:
```
{"status":403,"data":{"error":"Balloon\\Exception\\Forbidden","message":"not allowed to create new node here","code":38},"level":"info","message":"API REQ: Got body:","timestamp":"2017-11-19T10:51:15.227Z"}
{"level":"warning","message":"API: Balloon\\Exception\\Forbidden (not allowed to create new node here)","timestamp":"2017-11-19T10:51:15.228Z"}
{"level":"warning","message":"LOCALQUEUEHANDLER: Balloon\\Exception\\Forbidden (not allowed to create new node here) E_BLN_API_REQUEST_READ_ONLY_SHARE","timestamp":"2017-11-19T10:51:15.228Z"}
{"p":"/SLK-Operatives (Sitzungen, Organisation, Termine, Adressen, GoodPractice)","attributes":["share","access","readonly","id"],"level":"info","message":"API: GET request to /node/attributes with params: ","timestamp":"2017-11-19T10:51:
15.230Z"}
{"url":"/Applications/Balloon.app/Contents/Resources/app.asar/node_modules/@gyselroth/balloon-node-sync/lib/sync-db.js","line":272,"column":16,"error":{},"level":"error","message":"Uncaught SyntaxError: Invalid regular expression: /^/SLK-
Operatives (Sitzungen, Organisation, Termine, Adressen, GoodPractice)/Organisation (Adressen udn Zuständigkeiten(/.*|)$/: Unterminated group","timestamp":"2017-11-19T10:51:15.388Z"}
{"forceQuit":true,"level":"info","message":"Sync: STOP Requested","timestamp":"2017-11-19T10:51:22.882Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:28.577Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:29.074Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:34.832Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:35.310Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:41.093Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:41.578Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:47.382Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:47.707Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:48.418Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:48.905Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:53.386Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:53.649Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:54.692Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:54.961Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:51:59.430Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:51:59.921Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:52:00.759Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
","timestamp":"2017-11-19T10:52:01.266Z"}
{"level":"info","message":"Starting Sync with remote cursor: undefined","timestamp":"2017-11-19T10:52:05.734Z"}
{"message":"Sync: finished with error","stack":"Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","level":"error
```

**Should**:
Max. 5 helper tasks and no sync error.

#### balloon desktop version
0.0.31

#### Your client operating system
Mac OS X

**Comment by raffis** [comment]
error log:

`{"error":{"address":"xxx","code":"ENETUNREACH","errno":"ENETUNREACH","port":443,"syscall":"connect"},"level":"error","message":"Uncaught sync error. Resetting cursor and db","timestamp":"2017-11-16T16:49:19.129Z"}
{"level":"info","message":"Sync: not starting sync because it is already running","timestamp":"2017-11-16T16:49:19.132Z"}
{"level":"info","message":"Sync: ended","timestamp":"2017-11-16T16:49:19.132Z"}
{"level":"info","message":"Sync: starting sync","timestamp":"2017-11-16T16:49:49.132Z"}
{"level":"info","message":"Sync: starting sync","timestamp":"2017-11-16T16:49:49.133Z"}
{"level":"info","message":"Sync: pre sync check successfull","timestamp":"2017-11-16T16:49:49.134Z"}
{"level":"info","message":"Sync: pre sync check successfull","timestamp":"2017-11-16T16:49:49.194Z"}
{"error":{"address":"xxx","code":"ETIMEDOUT","errno":"ETIMEDOUT","port":443,"syscall":"connect"},"level":"error","message":"Uncaught sync error. Resetting cursor and db","timestamp":"2017-11-16T16:51:13.112Z"}
{"level":"info","message":"Sync: not starting sync because it is already running","timestamp":"2017-11-16T16:51:13.120Z"}
{"level":"info","message":"Sync: ended","timestamp":"2017-11-16T16:51:13.122Z"}
{"level":"info","message":"Sync: starting sync","timestamp":"2017-11-16T16:51:43.127Z"}
{"level":"info","message":"Sync: starting sync","timestamp":"2017-11-16T16:51:43.128Z"}
{"level":"info","message":"Sync: pre sync check successfull","timestamp":"2017-11-16T16:51:43.129Z"}
{"level":"info","message":"Sync: pre sync check successfull","timestamp":"2017-11-16T16:51:43.190Z"}
`

**Comment by raffis** [comment]
fixes regex issue in b1f42ad42b14cf00414052c48238c2a3b78dd2c0

**Comment by raffis** [comment]
`
Error: \"toString()\" failed\n    at Buffer.toString (buffer.js:503:11)\n    at tryToString (fs.js:445:15)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)","
`
is still an open issue, but the client probably shouldn't do a reset after receiving an error such as this.

**Comment by raffis** [comment]
`{"category":"bootstrap","error":{},"url":"/Applications/Balloon.app/Contents/Resources/app.asar/node_modules/@gyselroth/balloon-node-sync/lib/sync-db.js","line":272,"level":"error","message":"Uncaught sync error. Resetting cursor and db","timestamp":"2017-11-17T05:45:34.437Z"}
`
this error also happens to be in the error log various times before (But we do not have a sync error log entry yet cause of #35)

**Comment by raffis** [comment]
This is also related to #39

**Comment by raffis** [comment]
fixed with 0.0.34

**Comment by raffis** [comment]
probably still an issue #85
